common/functions/src/scalars/dates/number_function.rs

```diff
@@ -39,6 +39,7 @@ use crate::scalars::RoundFunction;
 #[derive(Clone, Debug)]
 pub struct NumberFunction<T, R> {
     display_name: String,
```

Comment:
Add the input_type is for?

Reply:
https://github.com/datafuselabs/databend/blob/b4c65cdfb47a59ad1d3bc300987527a75c4dcc5b/common/functions/src/scalars/dates/number_function.rs#L361-L369

For now the input type is infered during evaluation, which should be determined when instantiating the function.

Reply:
Currently, it's useless because we do not need it. But we may need it in the future.